src/pages-sections/addAllotment/addAllotment-form.jsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as yup from 'yup';
+import { Button, Card, Grid, MenuItem, TextField } from '@mui/material';
+import PropTypes from 'prop-types';
+
+const VALIDATION_SCHEMA = yup.object().shape({
+  name: yup.string().min(2).required('Name is required'),
+  bedno: yup.string().required('Bed no is required'),
+  startdate: yup.date().required('Start date is required'),
+  duedate: yup
+    .date()
+    .required('Due date is required')
+    .min(yup.ref('startdate'), 'Due date must be after start date'),
+  deposit: yup.number().required('Deposit is required'),
+  paymentmethod: yup.string().required('Payment method is required'),
+});
+const AddAllotmentForm = props => {
+  const { initialValues, handleFormSubmit } = props;
+  return (
+    <Card
+      sx={{
+        p: 6,
+        boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.5)',
+      }}
+    >
+      <Formik
+        onSubmit={handleFormSubmit}
+        initialValues={initialValues}
+        validationSchema={VALIDATION_SCHEMA}
+      >
+        {({
+          values,
+          errors,
+          touched,
+          handleChange,
+          handleBlur,
+          handleSubmit,
+        }) => (
+          <form onSubmit={handleSubmit}>
```

Comment:
Do we need this? we are handling submit with Formik right? no need to add html form element check and refactor this